cmd/spire-server/cli/run/run.go

```diff
@@ -525,6 +549,67 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 	return sc, nil
 }
 
+func parseBundleEndpointProfile(config federatesWithConfig) (trustDomainConfig *bundleClient.TrustDomainConfig, err error) {
+	var data bytes.Buffer
+	if err := printer.DefaultConfig.Fprint(&data, config.BundleEndpointProfile); err != nil {
+		return nil, err
+	}
+	configString := data.String()
```

Comment:
Looking at the config sample, I see we set bundle endpoint profile like this:
```hcl
federates_with "foo" {
    bundle_endpoint_url = "https://example.org/bundle.json"
    bundle_endpoint_profile "https_spiffe" {
        endpoint_spiffe_id = "spiffe://example.org/spire/server"
    }
}
```

While I think this is the best experience ergonomically, I also believe that it means that `bundle_endpoint_profile` is actually a string-indexed map. That means we can have a federates_with that has more than one `bundle_endpont_profile` declared? I don't see us handling this case here but I'm also having a hard time following all the nasty AST stuff so I might have just missed it :)

Assuming my read on the HCL semantics is correct, we should probably only allow `len(bundle_endpoint_profile) == 1`

Reply:
I wonder if we shouldn't do the following:
```
federates_with "foo" {
    bundle_endpoint_profile {
        https_spiffe {
            endpoint_spiffe_id = ..
        }
    }
}
```

Then we aren't messing around with strings at all, which can be misspelled...

Reply:
I have mixed feelings around this. I really like the shape of the config, but I also see it problematic because of #1101. I'm not sure if that issue will ever be fixed in HCL, or if the only option is to switch to HCL2.

I missed the check to ensure that only one profile is declared for a trust domain, I have to add that. I think that we will have to do that regardless of the shape of the config.

Reply:
I also like the shape of the existing thing, but wonder about the tradeoffs. On one hand, it is comfortable and is consistent with the rest of the config ... on the other hand, we have to do this weird AST stuff to accommodate a value that can be one of several types. I could go either way... maybe I weakly lean towards consistency.